tests/iso-ignition.sh

```diff
@@ -98,24 +98,27 @@ rm "${out_iso}"
 # Check that Ignition configs work independently of network configs
 echo "foo=baz" > one.nmconnection
 echo "bar=baz" > two.nmconnection
-coreos-installer iso network embed -k one.nmconnection -k two.nmconnection "${iso}"
-(coreos-installer iso ignition show "${iso}" 2>&1 ||:) | grep -q "No embedded Ignition config"
-coreos-installer iso ignition embed -i <(echo "${config}") "${iso}" -o "${out_iso}"
-coreos-installer iso ignition show "${out_iso}" | cmp - <(echo "${config}")
-coreos-installer iso network extract "${out_iso}" | grep -q "foo=baz"
-coreos-installer iso network extract "${out_iso}" | grep -q "bar=baz"
-coreos-installer iso ignition embed -i <(echo "${config}") "${iso}" -f
-coreos-installer iso network extract "${out_iso}" | grep -q "foo=baz"
-rm "${out_iso}"
-coreos-installer iso ignition remove "${iso}" -o "${out_iso}"
-coreos-installer iso network extract "${out_iso}" | grep -q "foo=baz"
-coreos-installer iso ignition remove "${iso}"
-coreos-installer iso network extract "${out_iso}" | grep -q "foo=baz"
-(coreos-installer iso ignition show "${iso}" 2>&1 ||:) | grep -q "No embedded Ignition config"
-coreos-installer iso network remove "${iso}"
-# verify we haven't written an empty cpio archive
-dd if="${iso}" skip="${offset}" count="${length}" bs=1 status=none | cmp -n "${length}" - /dev/zero
-rm "${out_iso}"
+if coreos-installer iso network embed -k one.nmconnection -k two.nmconnection "${iso}"; then
+    (coreos-installer iso ignition show "${iso}" 2>&1 ||:) | grep -q "No embedded Ignition config"
+    coreos-installer iso ignition embed -i <(echo "${config}") "${iso}" -o "${out_iso}"
+    coreos-installer iso ignition show "${out_iso}" | cmp - <(echo "${config}")
+    coreos-installer iso network extract "${out_iso}" | grep -q "foo=baz"
+    coreos-installer iso network extract "${out_iso}" | grep -q "bar=baz"
+    coreos-installer iso ignition embed -i <(echo "${config}") "${iso}" -f
+    coreos-installer iso network extract "${out_iso}" | grep -q "foo=baz"
+    rm "${out_iso}"
+    coreos-installer iso ignition remove "${iso}" -o "${out_iso}"
+    coreos-installer iso network extract "${out_iso}" | grep -q "foo=baz"
+    coreos-installer iso ignition remove "${iso}"
+    coreos-installer iso network extract "${out_iso}" | grep -q "foo=baz"
+    (coreos-installer iso ignition show "${iso}" 2>&1 ||:) | grep -q "No embedded Ignition config"
+    coreos-installer iso network remove "${iso}"
+    # verify we haven't written an empty cpio archive
+    dd if="${iso}" skip="${offset}" count="${length}" bs=1 status=none | cmp -n "${length}" - /dev/zero
+    rm "${out_iso}"
+else
+    echo "Failed to embed network settings; skipping"
```

Comment:
Hmm, maybe we should pass in to the test scripts the set of features we expect to be supported so we don't rely on feature detection itself not regressing.

Reply:
This isn't a blocker to be clear.

Reply:
Each of `iso-ignition.sh` and `iso-network.sh` test whether they work in the presence of the other feature.  If feature detection returns false negatives, `iso-network.sh` will catch it.  If it returns false positives, `unsupported.sh` should catch it, and anyway it doesn't reduce our test coverage here.

So I think it's okay to leave this as is.  That has the benefit that `iso-ignition.sh` can still be run by hand, without special knowledge.